enzyme/test/Enzyme/ProbProg/simple-condition.ll

```diff
@@ -0,0 +1,92 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -S | FileCheck %s
+
+@enzyme_condition = global i32 0
+
+@.str = private constant [3 x i8] c"mu\00"
+@.str.1 = private constant [2 x i8] c"x\00"
+
+declare double @normal(double, double)
+declare double @normal_logpdf(double, double, double)
+
+declare i8* @__enzyme_newtrace()
+declare void @__enzyme_freetrace(i8*)
+declare i8* @__enzyme_get_trace(i8*, i8*)
+declare i64 @__enzyme_get_choice(i8*, i8*, i8*, i64)
+declare void @__enzyme_insert_call(i8*, i8*, i8*)
+declare void @__enzyme_insert_choice(i8* %trace, i8*, double, i8*, i64)
+declare i1 @__enzyme_has_call(i8*, i8*)
+declare i1 @__enzyme_has_choice(i8*, i8*)
+declare double @__enzyme_sample(double (double, double)*, double (double, double, double)*, i8*, double, double)
+declare i8* @__enzyme_trace(void ()*)
+declare i8* @__enzyme_condition(void ()*, i32, i8*)
+
+define void @test() {
+entry:
+  %mu = call double @__enzyme_sample(double (double, double)* @normal, double (double, double, double)* @normal_logpdf, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @.str, i64 0, i64 0), double 0.0, double 1.0)
+  %x = call double @__enzyme_sample(double (double, double)* @normal, double (double, double, double)* @normal_logpdf, i8* getelementptr inbounds ([2 x i8], [2 x i8]* @.str.1, i64 0, i64 0), double %mu, double 1.0)
+  ret void
+}
+
+define i8* @generate() {
+entry:
+  %call = call i8* @__enzyme_trace(void ()* @test)
+  ret i8* %call
+}
+
+define i8* @condition(i8* %trace) {
+entry:
+  %0 = load i32, i32* @enzyme_condition
```

Comment:
These variable names being so similar will become confusing

Reply:
I am open to suggestions